src/main/java/games/strategy/engine/data/UnitCollection.java

```diff
@@ -257,4 +219,37 @@ public String toString() {
   public Iterator<Unit> iterator() {
     return Collections.unmodifiableList(m_units).iterator();
   }
+
+  @Override
+  public boolean contains(Object object) {
+    return m_units.contains(object);
+  }
+
+  @Override
+  public Object[] toArray() {
+    return m_units.toArray();
+  }
+
+  @Override
+  public <T> T[] toArray(T[] array) {
+    return m_units.toArray(array);
+  }
+
+  @Override
+  public boolean remove(Object object) {
+    final boolean result = m_units.remove(object);
+    m_holder.notifyChanged();
+    return result;
+  }
+
+  @Override
+  public boolean retainAll(Collection<?> collection) {
+    return m_units.retainAll(collection);
+  }
+
+  @Override
+  public void clear() {
+    m_units.clear();
+    m_holder.notifyChanged();
+  }
```

Comment:
Sure, but I can't find any instances where we're now passing a `UnitCollection` to these methods.  Is the plan to remove the `Match`-related methods in `UnitCollection` and replace those with calls to the same-named methods in `Match`?  For example, instead of calling `unitCollection.allMatch(match)` you'd call `Match.allMatch(unitCollection, match)`?